### ISSUE TITLE
Find a solution for L1, L2 loss (et al.)

### ISSUE THREAD

**Issue by ottonemo** [dnouri/skorch#81]
There should be an easy way to add Lx-regularization.

**Comment by ottonemo** [comment]
For regularization most optimizers implement L2 via the `weight_decay` parameter. This does not cover L1 and there is no way to apply the regularization term to specific parameters only. Another option is to implement this by adding an auxiliary term to the loss that computes the L* loss for a set of given parameters.

This begs the question where such a module would be implemented, as a separate class/module?

**Comment by benjamin-work** [comment]
About your code snippet:
The difficulty with that is that it requires the module to be initiated already when the net is defined. This is not how we encourage the user to use skorch. Furthermore, it wouldn't work for things like grid search, were the net is constantly re-initialized.

About letting the user implement it:
If I see it correctly, that should be rather easy (L1, untested):

```
class ReguralizedNet(NeuralNet):
    def get_loss(self, y_pred, y_true, X=None, train=False):
        loss = super().get_loss(y_pred, y_true, X=X, train=train)

        l1 = 0.01
        # only specific weight
        loss += l1 * self.module_.layer0.weight
        # all weights 
        loss += l1 * sum([w.abs().sum() for w in self.module_.parameters()])
        return loss
```

(We could add a special method that receives the loss and module after `get_loss` for this behavior, but that seems to be too much.)

The upside of not implementing Lx ourselves:
* (presumably) easy to implement yourself
* most common case already supported (L2 for all weights)
* it's a good way to teach the user to subclass our classes

Downside
* more work for the user
* code be be less performant

**Comment by ottonemo** [comment]
OK I'm in favor of *not* doing it ourselves but we should definitely document how L2 regularization can be applied. This is a common question for pytorch as well as it seems.

**Comment by benjamin-work** [comment]
The documentation now contains a section on L2 regularization in the FAQ and an example of how to override `get_loss` to add L1 regularization. I consider this enough, if not, please re-open.
